### PR TITLE
[Clean-up] Remove store abstraction

### PR DESCRIPTION
**Pull request by bbecquet**
## Description
Remove the abstraction level on store, which was used to expose local storage or Masq storage with the same API. Now `store.js` is the localStorage-based implementation.
Also uses a proper singleton implementation instead of using global vars.

*Another simplification would be to remove all these `async/await`, as the localstorage API is synchronous.*

## Why
Since the removal of Masq, the abstraction was useless. Let's remove some code complexity and shipped size.

### CODE REVIEW

File: src/adapters/store.js [hunk 1]
```diff
@@ -1,93 +1,111 @@
...
 import { getKey } from 'src/libs/pois';
 import { fire } from 'src/libs/customEvents';
+import { isPoiCompliantKey } from 'src/libs/pois';
 
-export default class Store {
-  constructor() {
-    // get store from window if already initialized
-    if (window.__store) {
-      return window.__store;
+class Store {
```
Comment (G-Ray) [inline src/adapters/store.js:8]: :+1
What do you think about even removing the class and exporting only functions ?

Comment (bbecquet) [reply src/adapters/store.js:8]: Yep, even simpler than a singleton, but I wanted to keep this PR with minimal changes outside the store file itself.
Do you want me to make the change here?

Comment (G-Ray) [reply src/adapters/store.js:8]: would be great !

Comment (bbecquet) [reply src/adapters/store.js:8]: Sure, doing it!

**Review by G-Ray — APPROVED** [review]

**Review by G-Ray — APPROVED** [review]

**Pull request merged**